src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -60,200 +113,828 @@ public static void init()
+            md5Map.remove(structureName.toString());
+            Log.getLogger().info("Structures: " + structureName + " deleted successfully");
+            return true;
+        }
+        else
+        {
+            Log.getLogger().warn("Failed to delete structure " + structureName);
+        }
+        return false;
+    }
+
+    /**
+     * check/create a directory and its parents.
+     *
+     * @param directory to be created
+     */
+    private static void checkDirectory(@NotNull final File directory)
+    {
+        if (!directory.exists() && !directory.mkdirs())
+        {
+            Log.getLogger().error("Directory doesn't exist and failed to be created: " + directory.toString());
+        }
+    }
+
+    /**
+     * Get the list of Sections.
+     * Builder, Citizen, Farmer ... + decorations and scans.
+     *
+     * @return list of sections.
+     */
+    @NotNull
+    @SideOnly(Side.CLIENT)
+    public static List<String> getSections()
+    {
+        final ArrayList<String> list = new ArrayList<>(schematicsMap.keySet());
+        Collections.sort(list);
+        return list;
+    }
+
+    /**
+     * Get the list of styles for a given section.
+     *
+     * @param section such as decorations, Builder ...
+     * @return the list of style for that section.
+     */
+    @NotNull
+    @SideOnly(Side.CLIENT)
+    public static List<String> getStylesFor(final String section)
+    {
+        if (schematicsMap.containsKey(section))
+        {
+            final Map<String, Map<String, String>> sectionMap = schematicsMap.get(section);
+            final ArrayList<String> list = new ArrayList<>(sectionMap.keySet());
+            Collections.sort(list);
+            return list;
         }
+        return new ArrayList<>();
+    }
 
-        final Integer level = hutLevelsMap.getOrDefault(hut, 0);
-        hutLevelsMap.put(hut, level + 1);
+    /**
+     * Get a list of schematics for this section and style.
+     *
+     * @param section such as Builder, schematics. scans ...
+     * @param style limit the list for schematics to this style.
+     * @return the list of schematics
+     */
+    @NotNull
+    @SideOnly(Side.CLIENT)
+    public static List<String> getSchematicsFor(final String section, final String style)
+    {
+        if (schematicsMap.containsKey(section))
+        {
+            final Map<String, Map<String, String>> sectionMap = schematicsMap.get(section);
+            if (sectionMap.containsKey(style))
+            {
+                final ArrayList<String> list = new ArrayList<>(sectionMap.get(style).values());
+                Collections.sort(list);
+                return list;
+            }
+        }
+        return new ArrayList<>();
     }
 
-    private static void addDecorationStyle(final String decoration, final String style)
+    /**
+     * Class to handle schematic naming.
+     * It does extract information from a schematic using its name.
+     */
+    public static class StructureName
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce this class from 118 to the maximum allowed 25 or externalize it in a public class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)